Project 2/k-NNAlgorithms/src/Data.java

```diff
@@ -1,2 +1,296 @@
+import java.io.File;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Data {
+    ArrayList<ArrayList<ArrayList<Double>>> splitData= new ArrayList<ArrayList<ArrayList<Double>>>();
+    ArrayList<ArrayList<Double>> fullSet = new ArrayList<ArrayList<Double>>();
+    String [][] inputFileAsArray;
+    String dataName;
+
+    Data(File inputFileName, String dataName) throws Exception{
+        this.dataName=dataName;
+        this.inputFileAsArray=fileTo2dString(inputFileName);
+        preprocessData();
+        convertInputFileToArrayList();
+        bucketizeData();
+    }
+
+    ///////////////////////////////////////////////////////////////
+    //function for input data to be changed into a 2d string array
+    private  String [][] fileTo2dString(File inputFile) throws Exception{
+
+        //determines the dimensions of the file
+        int count=0;
+        Scanner sc = new Scanner(inputFile);
+        while (sc.hasNextLine()){
```

Comment:
I want to eventually make this only scan through the file once. So don't forget to do that, Jay!